tests/test_imports.py

```diff
@@ -11,61 +13,145 @@
+#     """
+#     gold_mapper.unique_object_mapper()
+#     assert len(gold_mapper.nmdc_db.data_object_set) == 2
+#     assert len(gold_mapper.data_object_map) == 2
+
+
+# def test_multiple_object_mapper(gold_mapper):
+#     """
+#     This test counts the number of files from gold_mapper where the data object creation should be many:1.  JGI stores each binning file
+#     individually whereas NMDC combines all the records into a single tar.gz file.
+#     """
+#     gold_mapper.multiple_objects_mapper()
+#     # Add assertions to check if the method works as expected
+#     assert len(gold_mapper.nmdc_db.data_object_set) == 1
+#     print(gold_mapper.nmdc_db.data_object_set)
+#     assert len(gold_mapper.objects) == 1
+#     #check that the data object url gets made correctly for the multiple object mapper function.
+#     assert "https://data.microbiomedata.org/data/nmdc:omprc-11-importT/nmdc:abcd.1/nmdc_abcd.1_hqmq_bin.zip" in (do["url"] for do in gold_mapper.nmdc_db.data_object_set)
+
+def test_gold_mapper_map_sequencing_data(gold_mapper):
     """
-    Test the creation of workflow execution records and data objects that are has_output of those workflow execution subclasses.
+    Test that the gold mapper creates data objects for the sequencing data, and
+    provides an update to be applied to the has_output list of the sequencing data generation
     """
-    gold_mapper.unique_object_mapper()
-    gold_mapper.multiple_objects_mapper()
-    gold_mapper.workflow_execution_mapper()
-    gold_mapper.post_nmdc_database_object()
-    db = gold_mapper.get_database_object_dump()
-    #This should return 4 workflow_execution_set records becuase that is the number of records with Import:true in the config file
-    #note that if these records were tested against the actual schema they would fail b/c workflow executions can't have has_output be null.
-    assert len(db.workflow_execution_set) == 4
-    # gff files are 1:1 with data objects that are has_output of nmdc:MetagenomeAnnotation
-    # *tar.gz files should be combined into a single data object that is has_output of nmdc:MagsAnalysis
-    assert len(db.data_object_set) == 3
-
-
-def test_unique_object_mapper(gold_mapper):
+    exp_num_data_objects = 1    # There is only one sequencing data file from the gold import files fixture
+    exp_dobj_id = "nmdc:dobj-01-abcd1234"   # From the mock API minter response
+    exp_dobj_type = "Metagenome Raw Reads"  # From the gold_import_files fixture
+    exp_nucleotide_sequencing_id = "nmdc:omprc-11-importT"  # From the gold mapper fixture
+    exp_update = {
+        "collection": "data_generation_set",
+        "filter": {"id": exp_nucleotide_sequencing_id},
+        "update": {"has_output": [exp_dobj_id]}
+    }
+    # TODO verify that these are the correct values to expect based on the import logic for raw reads files
+    exp_url = 'https://data.microbiomedata.org/data/nmdc:omprc-11-importT/52834.4.466476.GATCGAGT-GATCGAGT.fastq.gz'
```

Comment:
in the past we've made data objects w/o urls for the output of DataGeneration since we aren't supposed to be a data repository, the data portal does not serve up the raw data for download.

Reply:
https://github.com/microbiomedata/nmdc_automation/pull/274/commits/5a58e578563b0f71c1351515219705210aad4ef1